hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/storage/TestBlockInputStream.java

```diff
@@ -174,20 +188,18 @@ public void testSeek() throws Exception {
   }
 
   @Test
-  public void testBlockEOF() throws Exception {
-    // Seek to some position < blockSize and verify EOF is not reached.
-    seekAndVerify(CHUNK_SIZE);
-    Assert.assertFalse(blockInputStream.blockStreamEOF());
-
-    // Seek to blockSize-1 and verify that EOF is not reached as the chunk
-    // has not been read from container yet.
-    seekAndVerify(blockSize-1);
-    Assert.assertFalse(blockInputStream.blockStreamEOF());
-  }
-
-  private void seekAndVerify(int pos) throws Exception {
-    blockInputStream.seek(pos);
-    Assert.assertEquals("Current position of buffer does not match with the " +
-            "seeked position", pos, blockInputStream.getPos());
+  public void testRead() throws Exception {
+    // read 200 bytes of data starting from position 50. Chunk0 contains
+    // indices 0 to 99, chunk1 from 100 to 199 and chunk3 from 200 to 299. So
+    // the read should result in 3 ChunkInputStream reads
+    seekAndVerify(50);
+    byte[] b = new byte[200];
+    blockStream.read(b, 0, 200);
+    matchWithInputData(b, 50, 250);
```

Comment:
Same here too, seek, read and continue with read, verify in the second read, we are reading continuing data or not?

Reply:
Added a new test testSeekAndRead()